File: integration-tests/iac/lambdas/python/s3_producer.py
import json
import os
import boto3

s3 = boto3.client('s3')


def handler(event, context):
    bucket_name = os.environ['BUCKET_NAME']

    body = {
        'message': 'Hello from S3 producer!',
        'requestId': context.aws_request_id,
    }

    key = f'test-events/{context.aws_request_id}.json'

    response = s3.put_object(
        Bucket=bucket_name,
        Key=key,
        Body=json.dumps(body),
        ContentType='application/json',
    )

    print(f"Put object to S3: {key}, ETag: {response['ETag']}")

    return {
        'statusCode': 200,
        'body': json.dumps({
            'key': key,
            'etag': response['ETag'],
        }),
    }

File: integration-tests/iac/lambdas/node/s3_producer.mjs
import { S3Client, PutObjectCommand } from '@aws-sdk/client-s3';

const s3 = new S3Client();

export async function handler(event, context) {
    const bucketName = process.env.BUCKET_NAME;

    const body = {
        message: 'Hello from S3 producer!',
        requestId: context.awsRequestId,
    };

    const key = `test-events/${context.awsRequestId}.json`;

    const response = await s3.send(new PutObjectCommand({
        Bucket: bucketName,
        Key: key,
        Body: JSON.stringify(body),
        ContentType: 'application/json',
    }));

    console.log(`Put object to S3: ${key}, ETag: ${response.ETag}`);

    return {
        statusCode: 200,
        body: JSON.stringify({
            key,
            etag: response.ETag,
        }),
    };
}
